libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -691,7 +687,13 @@ bool ContractCompiler::visit(WhileStatement const& _whileStatement)
 		m_context.appendConditionalJumpTo(loopEnd);
 	}
 
+	solAssert(m_loops.top() == &_whileStatement, "");
```

Comment:
What if `m_loops.empty()`? This and the next line would also be a good candidate for extracting into a function (I assume it is also used by other loops).

Reply:
Then it should definitely break :p 
Right, makes sense